eolearn/core/eodata_io.py

```diff
@@ -233,6 +233,15 @@ def _yield_savers(
         yield feature_saver
 
 
+def _get_feature_io_constructor(ftype: FeatureType, use_zarr: bool) -> type[FeatureIO]:
+    """Creates the correct FeatureIO, corresponding to the FeatureType."""
+    if ftype is FeatureType.META_INFO:
+        return FeatureIOJson
+    if ftype.is_vector():
+        return FeatureIOGeoDf
+    return FeatureIOZarr if use_zarr else FeatureIONumpy  # type: ignore[return-value] # not sure why
```

Comment:
This is probably unnecessary?

Reply:
mypy would tell us if it's not necessary :upside_down_face:  seems like some issue on their side when it comes to resolving inheritance hierarchies (wasnt an issue a while ago, not sure why it popped up?)

Reply:
My bad for not being clear. I thought that the whole function is not necessary, because you updated the code in `_get_feature_io` to be standalone, but I see now that it's used in `_yield_savers`

Reply:
ah yes, i also deleted it at first and was then surprised when tests started failing :see_no_evil: